sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -354,7 +354,17 @@ export class ServiceClient {
       }
 
       result = this.sendRequest(httpRequest)
-        .then(res => flattenResponse(res, operationSpec.responses[res.status]));
+        .then(res => flattenResponse(res, operationSpec.responses[res.status]))
+        .catch(ex =>
```

Comment:
I believe this `catch` will also catch errors coming from `flattenResponse` along with any coming from `sendRequest`

Our intention to flatten the error response should only be concerned with sendRequest right?
If so, then I would suggest the below

```javascript
try {
   result = await this.sendRequest(httpRequest);
} catch (err => {
 // flatten error response and throw
});
// flatten the result
}
```